patches/protonvpn-nm-lib/make-subprocess-wrapper-flatpak-aware.patch

```diff
@@ -0,0 +1,105 @@
+diff --git a/protonvpn_nm_lib/core/subprocess_wrapper.py b/protonvpn_nm_lib/core/subprocess_wrapper.py
+index 96634f6..9b2ed9e 100644
+--- a/protonvpn_nm_lib/core/subprocess_wrapper.py
++++ b/protonvpn_nm_lib/core/subprocess_wrapper.py
+@@ -1,5 +1,6 @@
+ import os
+ import platform
++import shutil
+ import subprocess as _subprocess
+ 
+ 
+@@ -25,9 +26,41 @@ class SubprocessWrapper:
+ 
+     @staticmethod
+     def is_root_owned(path):
++        if SubprocessWrapper.is_flatpak():
++            try:
++                owner_uid = int(_subprocess.check_output(
++                    ["flatpak-spawn", "--host", "stat", "-c", "%u", path]).decode().strip())
++                owner_gid = int(_subprocess.check_output(
++                    ["flatpak-spawn", "--host", "stat", "-c", "%g", path]).decode().strip())
++                return owner_uid == owner_gid == 0
++            except _subprocess.CalledProcessError:
++                pass
+         stat_info = os.stat(path)
+         return stat_info.st_uid == 0 and stat_info.st_gid == 0
+ 
++    @staticmethod
++    def is_flatpak():
++        """Checks if it's in the Flatpak sandbox environment.
++        """
++        return os.path.isfile("/.flatpak-info")
++
++    @staticmethod
++    def which(binary):
++        """Returns the path to an executable which would be run if the given cmd was called.
++
++        If we are in a Flatpak, get the corresponding host command.
++        """
++        if SubprocessWrapper.is_flatpak():
++            try:
++                return _subprocess.check_output(["flatpak-spawn", "--host", "which", binary]).decode().strip()
```

Comment:
What are you trying to run on the host and why?

Reply:
I understand you aren't upstream but it should be talking to `org.freedesktop.NetworkManager` and systemd's equivalent dbus. This is a terrible permission.

Reply:
They may limit their own usage but its still a sandbox escape.

I would at least start a discussion with upstream so they know that the DBus APIs are a better way to do this.

Reply:
xdg-open should not be called outside the sandbox anyway.

Reply:
Hmm, I wonder if `nmcli` actually can use DBus itself. Have you tried building it inside of the package? Looking through the codebase its full of DBus usage.